src/app/reporting/Engine.cpp

```diff
@@ -244,6 +275,13 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
     err = reportDataBuilder.Init(&reportDataWriter);
     SuccessOrExit(err);
 
+    if (apReadHandler->IsSubscription())
+    {
+        uint64_t subscriptionId = 0;
+        apReadHandler->GetSubscriptionId(subscriptionId);
```

Comment:
does GetSubscriptionId return a CHIP_ERROR? if so, it should be handled.

Reply:
no error, thanks